addons/udes_stock/tests/test_picking_batch.py

```diff
@@ -834,6 +834,178 @@ def test27_unpickable_item_product_ok_two_picks(self):
+                 when unpickable of one of the pickings will create an
+                 investigation of 3.
+        """
+        Picking = self.env['stock.picking']
+        Batch = self.env['stock.picking.batch']
+        Batch = Batch.sudo(self.outbound_user)
+
+        quant = self.create_quant(self.apple.id, self.test_location_01.id, 4)
+
+        products_info = [{'product': self.apple, 'qty': 1}]
+        picking = self.create_picking(self.picking_type_pick,
+                                      products_info=products_info,
+                                      confirm=True,
+                                      assign=True)
+
+        picking_2 = self.create_picking(self.picking_type_pick,
+                                        products_info=products_info,
+                                        confirm=True,
+                                        assign=True)
+        pickings = picking | picking_2
+        batch = Batch.create_batch(self.picking_type_pick.id, None)
+        batch.add_extra_pickings(self.picking_type_pick.id)
+        self.assertEqual(pickings, batch.picking_ids)
+
+        self.create_picking(self.picking_type_pick,
+                            products_info=products_info,
+                            confirm=True,
+                            assign=True)
+        # check that now 3 units of the quant are reserved
+        self.assertEqual(quant.reserved_quantity, 3)
+
+        reason = 'missing item'
+        move_line = picking.move_line_ids[0]
+
+        batch.unpickable_item(product_id=move_line.product_id.id,
+                              location_id=move_line.location_id.id,
+                              reason=reason)
+        # after unpickable all the quant should be reserved
+        self.assertEqual(quant.reserved_quantity, 4)
+        # picking state should be confirmed
+        self.assertEqual(pickings.mapped('state'), ['confirmed', 'confirmed'])
+        # Check no backorder has been created
+        self.assertEqual(len(pickings.filtered('u_created_back_orders')), 0)
+        # check that the investigation has reserved 3 only
+        inv_picking = Picking.search([('picking_type_id', '=', self.picking_type_internal.id)])
+        self.assertEqual(len(inv_picking), 1)
+        self.assertEqual(len(inv_picking.move_line_ids), 1)
+        self.assertEqual(inv_picking.move_line_ids[0].product_qty, 3)
+
+    def test30_backorder_generated_unpickable_item_product_ok_in_two_picks_and_two_products(self):
+        """
+        Tests that calling unpickable item with a product in two pickings, one
+        of which also includes a line for another product, preserves the move
+        for the pickable product.
+        """
+        Picking = self.env['stock.picking']
+        Batch = self.env['stock.picking.batch']
+        Batch = Batch.sudo(self.outbound_user)
+
+        quant = self.create_quant(self.apple.id, self.test_location_01.id, 4)
```

Comment:
Nit: Could make the variables more descriptive, e.g. `apple_quant` and `banana_quant`. Same for products info and picking variables.

Reply:
I've removed the `_2` suffix and prefixed those variable names with `mixed_` -  I think what we want to express is that there is more than one product involved, rather than the specific product names.  Thanks for the nudge to make this improvement.